controllers/C_Document.class.php

```diff
@@ -158,6 +158,14 @@ function upload_action_process() {
         $sentUploadStatus = array();
         if (count($_FILES['file']['name']) > 0) {
             $upl_inc = 0;
+            for($i = 0; $i < count($_FILES['file']['name']); $i++){
+                $allowed = array('php','cpp','java','json','bat','exe','js','asp','asx','cfm','yaws','sh','py','rb','dll','cgi');
```

Comment:
cpp has to be compiled to do anything.  Perl extensions are a good one to put in for that spot, though the browser cannot directly execute them.  Screensavers are another good one.
Still, a pretty good list.  We should do another allow/disallow feature for handling the $allowed list here in the future.
Take care of removing the sqlconf.php from the PR.  This does not mean deleting the file from the repo.  Just take it out of the commit (drop the changes).  You can probably do that with another commit that just has a copy of the original.  If you like, change it by adding a comment like "Attention: exclude this file in code repository operations" or something like that to sort of bypass this whole mess.  :)